bundler/spec/runtime/inline_spec.rb

```diff
@@ -355,6 +355,26 @@ def confirm(msg, newline = nil)
     expect(out).to include("BUNDLE_GEMFILE is empty")
   end
 
+  it "uses already activated specs if they satisfy dependencies" do
+    script <<-RUBY
+      gemfile do
+        source "#{file_uri_for(gem_repo1)}"
+        gem "rack", "0.9.1"
+      end
+
+      gemfile(true) do
+        source "#{file_uri_for(gem_repo1)}"
+        gem "rack", "<= 1.0.0"
+      end
```

Comment:
`gemfile(true)` shouldn't be necessary here, but when using only `gemfile` (no `(true)`), I can't get Bundler to (re)resolve `gem "rack", "<= 1.0.0"` to `1.0.0`.  (i.e. It resolves to `0.9.1` even when `allow_compatible_activated_specs` is false.)